crates/rethnet_eth/src/transaction/signed/eip155.rs

```diff
@@ -0,0 +1,86 @@
+use bytes::Bytes;
+use revm_primitives::{keccak256, Address, B256, U256};
+
+use crate::{
+    signature::{Signature, SignatureError},
+    transaction::{kind::TransactionKind, request::EIP155TransactionRequest},
+};
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+#[cfg_attr(
+    feature = "fastrlp",
+    derive(open_fastrlp::RlpEncodable, open_fastrlp::RlpDecodable)
+)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct EIP155SignedTransaction {
+    pub nonce: u64,
+    pub gas_price: U256,
+    pub gas_limit: u64,
+    pub kind: TransactionKind,
+    pub value: U256,
+    pub input: Bytes,
+    pub signature: Signature,
+}
+
+impl EIP155SignedTransaction {
+    pub fn hash(&self) -> B256 {
+        keccak256(&rlp::encode(self))
+    }
+
+    /// Recovers the Ethereum address which was used to sign the transaction.
+    pub fn recover(&self) -> Result<Address, SignatureError> {
+        self.signature
+            .recover(EIP155TransactionRequest::from(self).hash())
+    }
+
+    pub fn chain_id(&self) -> u64 {
+        (self.signature.v - 35) / 2
+    }
+
+    /// See <https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md>
+    /// > If you do, then the v of the signature MUST be set to {0,1} + `CHAIN_ID` * 2 + 35 where
+    /// > {0,1} is the parity of the y value of the curve point for which r is the x-value in the
+    /// > secp256k1 signing process.
+    pub fn meets_eip155(&self, chain_id: u64) -> bool {
```

Comment:
Should we reject `chain_id == 0`? Would be good to have some tests here as well for edge cases

Reply:
I removed these functions in https://github.com/NomicFoundation/hardhat/pull/4298/commits/99653bcb79aee565cae47c479caa4a7c03121da9, as they are no longer necessary.